common/xtNetworkRequestManager.cpp

```
@@ -0,0 +1,77 @@
+/*
+ * This file is part of the xTuple ERP: PostBooks Edition, a free and
+ * open source Enterprise Resource Planning software suite,
+ * Copyright (c) 1999-2015 by OpenMFG LLC, d/b/a xTuple.
+ * It is licensed to you under the Common Public Attribution License
+ * version 1.0, the full text of which (including xTuple-specific Exhibits)
+ * is available at www.xtuple.com/CPAL.  By using this software, you agree
+ * to be bound by its terms.
+ */
+
+#include "xtNetworkRequestManager.h"
+#include <QNetworkAccessManager>
+#include <QNetworkRequest>
+#include <QNetworkReply>
+#include <QMutex>
+#include <QMutexLocker>
+#include <QObject>
+#include <QDebug>
+
+#define DEBUG false
+
+xtNetworkRequestManager::xtNetworkRequestManager(const QUrl & url, QMutex &mutex) {
+  nwam = new QNetworkAccessManager;
+  _nwrep = 0;
+  _response = 0;
+  _url = url;
+  _mutex = &mutex;
+  _mutex->lock();
+  _loop = new QEventLoop;
+  startRequest(_url);
+}
+void xtNetworkRequestManager::startRequest(const QUrl & url) {
+    _nwrep = nwam->get(QNetworkRequest(url));
+    connect(_nwrep, SIGNAL(finished()), SLOT(requestCompleted()));
+    connect(nwam, SIGNAL(sslErrors(QNetworkReply*,QList<QSslError>)), this, SLOT(sslErrors(QNetworkReply*,QList<QSslError>)));
+#if QT_VERSION < 0x050000 //this feels hackish, ignore error or provide root cert, we'll ignore for now
+    connect(nwam, SIGNAL(sslErrors(QNetworkReply*,QList<QSslError>)), _nwrep, SLOT(ignoreSslErrors()));
```

It's a hack-ish way around the SSL error encountered on the https version of the site, ignore the error and don't stop execution, rather than providing a proper cert for the connection(until we implement a lasting solution for qt5)